include/pagmo/problems/unconstrain.hpp

```diff
@@ -0,0 +1,379 @@
+    void set_c_tol(const vector_double &) = delete;
+    vector_double get_c_tol() const = delete;
+    bool feasibility_x(const vector_double &) const = delete;
+    bool feasibility_f(const vector_double &) const = delete;
+
+// The CI using gcc 4.8 fails to compile this delete, excluding it in that case does not harm
+// it would just result in a "weird" behaviour in case the user would try to stream this object
+#if __GNUC__ > 4
+    // NOTE: We delete the streaming operator overload called with unconstrain, otherwise the inner prob would stream
+    // NOTE: If a streaming operator is wanted for this class remove the line below and implement it
+    friend std::ostream &operator<<(std::ostream &, const unconstrain &) = delete;
+#endif
+    template <typename Archive>
+    void save(Archive &) const = delete;
+    template <typename Archive>
+    void load(Archive &) = delete;
+
+    // B - Specific to the unconstrain
+    // An unconstrained problem does not have gradients (most methods are not differentiable)
+    vector_double gradient(const vector_double &dv) const = delete;
+    // deleting has_gradient allows the automatic detection of gradients to see that unconstrain does not have any
+    // regardless of whether the class its build from has them. A unconstrain problem will thus never have gradients
+    bool has_gradient() const = delete;
+    // deleting has_gradient_sparsity/gradient_sparsity allows the automatic detection of gradient_sparsity to see that
+    // unconstrain does have an implementation for it. The sparsity will thus always be dense and referred to a problem
+    // with one objective
+    bool has_gradient_sparsity() const = delete;
+    sparsity_pattern gradient_sparsity() const = delete;
+    // An unconstrained problem does not have hessians (most methods are not differentiable)
+    std::vector<vector_double> hessians(const vector_double &dv) const = delete;
+    // deleting has_hessians allows the automatic detection of hessians to see that unconstrain does not have any
+    // regardless of whether the class its build from has them. An unconstrained problem will thus never have hessians
+    bool has_hessians() const = delete;
+    // deleting has_hessians_sparsity/hessians_sparsity allows the automatic detection of hessians_sparsity to see that
+    // unconstrain does have an implementation for it. The hessians_sparsity will thus always be dense
+    // (unconstrain::hessians_sparsity) and referred to a problem with one objective
+    bool has_hessians_sparsity() const = delete;
+    std::vector<sparsity_pattern> hessians_sparsity() const = delete;
+    // No need for these the default implementation of these in pagmo::problem already returns zero.
+    vector_double::size_type get_nec() const = delete;
+    vector_double::size_type get_nic() const = delete;
+
+    /// types of unconstrain methods
+    enum class method_type {
+        DEATH = 0,    ///< "death penalty"
```

Comment:
and in any case implicit conversion between the enums value and the int is not allowed anymore right?